tracks/python/exercises/matrix/mentoring.md

```diff
@@ -0,0 +1,57 @@
+### Problem and challenges
+
+The problem asks you to write a Matrix class.
+The constructor takes as input a string holding 
+lines of integers. The problem asks you
+to write methods row() and column().
+
+### Reasonable Solution
+
+```python
+class Matrix(object):
+
+    def __init__(self, text: str):
+        self.rows = [[int(s) for s in word.split()]
```

Comment:
@cmccandless  - .split() in this example is removing whitespace.  The digits in the rows of the matrix string are space delimited. 

@jeffdparker  - strongly recommend that the first Solution  in this section be the one from https://github.com/exercism/python/blob/master/exercises/matrix/example.py, which can then be followed by other variants.   The extemporaneous use of `.copy()` in the current example is a little problematic, IMO.

Reply:
> .split() in this example is removing whitespace. The digits in the rows of the matrix string are space delimited.

:+1: thanks for that; I wasn't looking at the test suite or README when I wrote that, and as I always try to provide an argument for `str.split()`, I forgot the default was `' '`, not `''` (I've been doing more JS lately, which has empty string for its default in string split).

> The extemporaneous use of .copy() in the current example is a little problematic, IMO.

Respectfully, I disagree. I think it is cleaner, despite being an extra step. The test suite does not test for the following case, but in the real world, it would be a concern if a user could modify the original matrix with the following:
```Python
m = Matrix('1 2\n3 4')
m.rows[0][0] = 2
```

Exercism is meant for teaching fluency in a language, and that includes best practices. I would certainly not discourage the use of `.copy()` (I might even praise the foresight of students who include it), but that said, I would not withhold approval from solutions that didn't use it (or equivalent code).

Reply:
The use of `.copy()` in the `rows()` function doesn't prevent modification of  `self.rows` directly in any way.  Granted, it _does_ prevent the caller from inadvertently mutating by:

```python
m = Matrix('1 2\n3 4')
foo = m.rows(0)
foo[0] = 7
```
Which is a good consideration.  I think I am more used to seeing attributes protected via `@property` than via copying.